File: src/main/java/com/nonxedy/model/Cutscene.java
package com.nonxedy.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Represents a cutscene with a name and a list of frames
 */
public class Cutscene {
    private final String name;
    private final List<CutsceneFrame> frames;
    
    /**
     * Creates a new cutscene
     * @param name The name of the cutscene
     * @param frames The frames of the cutscene
     */
    public Cutscene(String name, List<CutsceneFrame> frames) {
        this.name = name;
        this.frames = new ArrayList<>(frames);
    }
    
    /**
     * Gets the name of the cutscene
     * @return The name
     */
    public String getName() {
        return name;
    }
    
    /**
     * Gets the frames of the cutscene
     * @return The frames
     */
    public List<CutsceneFrame> getFrames() {
        return frames;
    }
}

File: bin/main/com/nonxedy/model/Cutscene.kt
package com.nonxedy.model

import kotlin.math.ceil

// Represents a cutscene with a name and a list of frames
data class Cutscene(
    val name: String,
    val frames: List<CutsceneFrame>,
    val frameDurationMs: Long = 50L
) {
    constructor(name: String, frames: List<CutsceneFrame>, ticksPerFrame: Int) : this(
        name,
        frames,
        ticksPerFrame.coerceAtLeast(1) * 50L
    )

    val ticksPerFrame: Int
        get() = ceil(frameDurationMs.coerceAtLeast(1L) / 50.0).toInt()
}
